Firestore/core/src/core/filter.cc

```diff
@@ -32,6 +35,11 @@ std::ostream& operator<<(std::ostream& os, const Filter& filter) {
   return os << filter.ToString();
 }
 
+Filter::Rep::Rep()
```

Comment:
can I initialize the `memoized_flattened_filters_` directly in `filter.h` instead of here in Rep constructor?  like `memoized_normalized_order_bys_`

Reply:
You can try. The issue was that `memoized_flattened_filters_` uses `FieldFilter` as its type parameter, which is a subclass of the class currently being defined. So there was a bit of a chicken-and-egg problem. By moving the initialization to the .cc file it worked around this circular dependency.